### PR TITLE
feat(core-flows,medusa,types): add automatic-taxes to region + generate tax lines endpoint

### PR DESCRIPTION
**Pull request by riqwan**
what:

- endpoint to generate tax lines
- update workflows to force calculate tax lines with a flag
- added automatic_taxes to region

### CODE REVIEW

File: packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts [hunk 1]
```diff
@@ -9,21 +9,36 @@ import {
...
 ): TaxCalculationContext | null {
   const address = cart.shipping_address
+  const shouldCalculateTax = forceTaxCalculation || cart.region?.automatic_taxes
+
+  if (!shouldCalculateTax) {
+    return null
+  }
+
+  if (!address || !address?.country_code) {
+    if (forceTaxCalculation) {
```
Comment (carlos-r-l-rodrigues) [inline packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts]:
```suggestion
  if (forceTaxCalculation && !address?.country_code) {
```

**Review by olivermrbl — APPROVED** [review]
Code-wise, LGTM!

File: packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts (hunk 1 above)
Comment (riqwan) [reply packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts]: @carlos-r-l-rodrigues we want this to skip if there is no country code and only throw if its forcing a calculation

Comment (olivermrbl) [reply packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts]: ~~I think carlos is just suggesting a simplification of the check no?~~

My bad, I see what you mean.

Comment (riqwan) [reply packages/core-flows/src/definition/cart/steps/get-item-tax-lines.ts]: doing this breaks the flow of `forceTaxCalculation=false` to return with null if address.country_code is not present. We could remove the `!address` or break it into 2 if conditions

This is the expected flow:
```
if (forceTaxCalculation && !countryCode) {
  throw error
}

if (!countryCode) {
  return null
}
```

**Review by sradevski — APPROVED** [review]

**Pull request merged**